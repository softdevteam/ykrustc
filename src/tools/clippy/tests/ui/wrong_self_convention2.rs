// edition:2018
#![warn(clippy::wrong_self_convention)]
#![warn(clippy::wrong_pub_self_convention)]
#![allow(dead_code)]

fn main() {}

mod issue6983 {
    pub struct Thing;
    pub trait Trait {
        fn to_thing(&self) -> Thing;
    }

    impl Trait for u8 {
        // don't trigger, e.g. `ToString` from `std` requires `&self`
        fn to_thing(&self) -> Thing {
            Thing
        }
    }

    trait ToU64 {
        fn to_u64(self) -> u64;
    }

    struct FooNoCopy;
    // trigger lint
    impl ToU64 for FooNoCopy {
        fn to_u64(self) -> u64 {
            2
        }
    }
}
