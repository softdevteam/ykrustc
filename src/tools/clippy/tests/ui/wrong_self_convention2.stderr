error: methods with the following characteristics: (`to_*` and `self` type is not `Copy`) usually take `self` by reference
  --> $DIR/wrong_self_convention2.rs:28:19
   |
LL |         fn to_u64(self) -> u64 {
   |                   ^^^^
   |
   = note: `-D clippy::wrong-self-convention` implied by `-D warnings`
   = help: consider choosing a less ambiguous name

error: aborting due to previous error

